File: JavaScript/740-Delete and Earn.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var deleteAndEarn = function(nums) {
    let hmap = new Map();
    let skip = 0, using = 0, prev = null;
    for (let num of nums){
        hmap.set(num, (hmap.get(num)||0) + 1);
    }
    sortedNum = [...hmap.keys()].sort(function(a,b){
        return a-b;
    })
    
    for (let num of sortedNum){
        if (num-1 !== prev){
            [skip, using] = [Math.max(skip, using), Math.max(skip, using) + num*hmap.get(num)];
        }else{
            [skip, using] = [Math.max(skip, using), skip + num*hmap.get(num)];
        }
        prev = num;
    }
    return Math.max(skip, using);
};

File: 740-Delete and Earn.py
class Solution(object):
    def deleteAndEarn(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        hmap = collections.Counter(nums)
        prev = None
        skip = has = 0
        
        for num in sorted(hmap.keys()):
            if num-1 != prev:
                skip, has = max(skip, has), max(skip, has) + num*hmap[num]
            else:
                skip, has = max(skip, has), skip + num*hmap[num]
            prev = num
            
        return max(skip, has)